testing/test_version.py

```diff
@@ -37,3 +38,17 @@
 def test_next_semver(version, expected_next):
     computed = simplified_semver_version(version)
     assert computed == expected_next
+
+
+@pytest.mark.parametrize(
+    "tag, expected",
+    [
+        pytest.param("v1.0.0", "1.0.0"),
+        pytest.param("v1.0.0-rc.1", "1.0.0rc1"),
+        pytest.param("v1.0.0-rc.1+-25259o4382757gjurh54", "1.0.0rc1"),
+    ],
+)
+def test_tag_regex1(tag, expected):
+    Configuration().tag_regex = r'^(?P<prefix>v)?(?P<version>[^\+]+)(?P<suffix>.*)?$'
```

Comment:
the reason why the singleton is not acceptable demonstrates in this test - if it was executed as first test, - all tests would use the different configuration

Reply:
You are right. Would it be acceptable to keep the class instance and push it through all the way from get_version to meta instead?

Reply:
that seems to be the only way (and one of the reasons why i avoided implementing myself

also its not clear if that config object should be everywhere

Reply:
Had a look into it. Not that difficult, but configuration needs to be passed to any scanner's `parse` entry points on to the version parser `meta`. Parse doesn't have kwargs. Even if optional it would be a change to an API your exposing. Config would be made available to both scanners and parser.

Separating parsers/scanners by passing back scanner results and then parse them from the main routine would probably the better approach from an architectural view. That would require even more refactoring but would allow keep the entry points stable until you would want to configure those further.

I was thinking about doing a trimmed down fork to only read version information borrowing your scanners as an alternative. Versioning being handled by the build-systems seems to be the preferred approach today.

Reply:
im under the impression that some more design work is needed

note that setuptools_scm is intended as a entry in the build-systems requires table ^^